3p-ecosystem/src/main/java/com/google/homesampleapp/screens/home/HomeFragment.kt

```diff
@@ -409,16 +439,56 @@ class HomeFragment : Fragment() {
     }
   }
 
+  // -----------------------------------------------------------------------------------------------
+  // Composables
+
+  @Composable
+  private fun HomeScreen(homeViewModel: HomeViewModel) {
+    // Observes values coming from the VM's devicesUiModelLiveData
+    val devicesUiModel by homeViewModel.devicesUiModelLiveData.observeAsState()
+    Timber.d("HomeScreen [${devicesUiModel}]")
+    if (devicesUiModel == null || devicesUiModel!!.devices.isEmpty()) {
+      NoDevices()
+    }
+  }
+
+  @Composable
+  private fun NoDevices() {
+    Column {
+      Image(
+        painter = painterResource(R.drawable.emptystate_missing_content),
+        contentDescription = stringResource(R.string.no_devices_image),
+        modifier = Modifier.fillMaxWidth().height(200.dp)
+      )
+      Text(
+        text = stringResource(R.string.no_devices_yet),
+        style = MaterialTheme.typography.subtitle1,
+        modifier = Modifier.fillMaxWidth().wrapContentWidth(Alignment.CenterHorizontally)
+      )
+      Text(
+        text = stringResource(R.string.add_your_first),
+        style = MaterialTheme.typography.subtitle2,
+        modifier = Modifier.fillMaxWidth().wrapContentWidth(Alignment.CenterHorizontally)
+      )
+    }
+  }
+
+  @Preview
+  @Composable
+  private fun NoDevicesPreview() {
+    MaterialTheme {
+      NoDevices()
+    }
+  }
+
   // -----------------------------------------------------------------------------------------------
   // UI Updates
 
   private fun updateUi(devicesUiModel: DevicesUiModel) {
```

Comment:
It'll be sooooooo nice when the list is converted to Compose and we don't need things like this... ;)

Reply:
agree :-)